Sources/PointFree/Account/Invite.swift

```diff
@@ -0,0 +1,137 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+import UrlFormEncoding
+
+let showInviteMiddleware =
+  requireTeamInvite
+    <| currentUserMiddleware
+    >-> writeStatus(.ok)
+    >-> respond(showInviteView)
+
+let revokeInviteMiddleware =
+  requireTeamInvite
+    <<< requireUser
+    <| { conn in
+      AppEnvironment.current.database.deleteTeamInvite(get2(conn.data).id)
+        .run
+        .flatMap(const(conn |> redirect(to: path(to: .team))))
+}
+
+let resendInviteMiddleware =
+  requireTeamInvite
+    <<< requireUser
+    <| { conn in
+      parallel(sendInviteEmail(invite: get2(conn.data), inviter: get1(conn.data)).run)
+        .run({ _ in })
+      return conn |> redirect(to: path(to: .team))
+}
+
+let acceptInviteMiddleware =
+  requireTeamInvite <| { conn in
+    
+    // TODO: subscribe user
+    // TOOD: send email to inviter
+    conn
+      |> redirect(to: path(to: .account))
+    
+}
+
+let sendInviteMiddleware =
+  requireUser
+    <| { (conn: Conn<StatusLineOpen, Tuple2<Database.User, EmailAddress?>>) in
+
+      guard let email = get2 <| conn.data else { return conn |> redirect(to: path(to: .team)) }
+      let inviter = get1 <| conn.data
+
+      return AppEnvironment.current.database.insertTeamInvite(email, inviter.id)
+        .run
+        .flatMap { errorOrTeamInvite in
+          switch errorOrTeamInvite {
+          case .left:
+            return conn |> redirect(to: .team)
+
+          case let .right(invite):
+            parallel(sendInviteEmail(invite: invite, inviter: inviter).run)
+              .run({ _ in })
+
+            return conn |> redirect(to: .team)
+          }
+      }
+}
+
+private let showInviteView = View<Tuple2<Database.User?, Database.TeamInvite>> { data in
+
+  get1(data)
+    .map { showInviteLoggedInView.view($0 .*. get2(data)) }
+    ?? showInviteLoggedOutView.view(get2(data))
+}
+
+private let showInviteLoggedOutView = View<Database.TeamInvite> { invite in
+  [
+    p(["You must be logged in to accept this invitation. Would you like to log in with GitHub?"]),
```

Comment:
Should we just redirect immediately?

Reply:
I thought it would be nice to let them know why we are asking for GH auth first